File: src/WebSocketManager.Common/InvocationDescriptor.cs
namespace WebSocketManager.Common
{
    public class InvocationDescriptor : InvocationMessage
    {
        public string MethodName { get; set; }

        public object[] Arguments { get; set; }

        public override string ToString()
        {
            return $"{MethodName}, {Arguments?.Length ?? 0} Arguments";
        }
    }
}

File: src/MorseL.Client.TS/src/InvocationDescriptor.ts
export class InvocationDescriptor {
    private static CurrentId = 1;
    public Id: number;
    public MethodName: string;
    public Arguments: Array<any>;

    constructor(methodName: string, args: any[]) {
        this.Id = InvocationDescriptor.CurrentId++;
        this.MethodName = methodName;
        this.Arguments = args;
    }
}